ocaml/xapi/xapi_http.ml

```diff
@@ -316,3 +316,8 @@ let add_handler (name, handler) =
       | Datamodel.Options -> Http.Options
     in Http_svr.Server.add_handler server ty uri h
 
+  let task_id_of_req (req: Http.Request.t) = 
+    let all = Http.Request.(req.cookie @ req.query) in
+    if List.mem_assoc "task_id" all
+    then Some (Ref.of_string (List.assoc "task_id" all))
+    else None
```

Comment:
The second, refactoring commit could also use this inside `Xapi_http.assert_credentials_ok` and `Xapi_http.with_context`.